src/QirRuntime/README.md

```diff
@@ -0,0 +1,115 @@
+# The Native Quantum Runtime [incubation project]
+
+This project contains headers for the native runtime, [possibly incomplete] implementations of a few
+simulators, and the QIR-to-native bridge to support running QIR files against simulators from this or other native runtime(s).
+
+- `public` folder contains the public headers
+- `lib` folder contains the implementation of the runtime, the simulators and the QIR bridge
+- `test` folder contains tests for the runtime
+- `experimental` folder contains features for which specifications and design are in too early stage to include into the QDK.
+
+## Build
+
+The project is using CMake (3.17) + Ninja(1.10.0) + Clang++(10.0.0). Other versions of the tools might work but haven't been tested. Only x64 architecture is supported.
+
+You can use CMake directly. For example, to produce Release build:
+
+1. navigate into the project's root folder
+2. mkdir build
+3. cd build
+4. cmake -G Ninja -DCMAKE_BUILD_TYPE=Release ..
+5. cmake --build .
+
+Or you can run `build.py` script from the root of the project. The default options for the script are `make debug`.
+
+- (Windows) `python build.py [make/nomake] [debug|release] [ir]`
+- (Linux) `python3 build.py [make/nomake] [debug|release] [ir]`
+
+The scrip will place the build artefacts into `build/[Windows|Linux]/[Debug|Release]` folder.
+
+### Windows pre-reqs
+
+1. Install Clang, Ninja and CMake from the public distros.
+2. Add all three to your/system `%PATH%`.
+3. Install VS 2019 (Clang uses MSVC's standard library on Windows).
+4. [_optional_]
+    1. Either clone LLVM's repo and build LLVM locally to produce your own `opt.exe` and other IR tools or download LLVM's dev package.
+    2. Either modify ProduceIR.cmake to point to these tools, or copy them into a separate folder and add that folder to `%PATH%.` Don't add the whole target build folder to `%PATH%` because it would likely confuse Clang.
+    3. To produce IR targets automatically as part of the build, either pass `ir` option to `build.py` or uncomment setting `${GENERATE_IR}` variable in the root `CMakeLists.txt`.
+
+*Building from Visual Studio and VS Code is **not** supported.
+Running cmake from the editors will likely default to MSVC or clang-cl and fail.*
+
+### Linux via WSL pre-reqs
+
+1. On the host Windows machine install WLS.
+2. In WSL install clang/llvm and the other tools:
+    1. NB: the default version of clang/llvm on WSL's Ubuntu are likely to be too old, it's recommended you get the latest packages.
+    2. $ sudo apt install clang
```

Comment:
Thanks for the detailed instructions! I'm starting off my review by trying out the local compile and test. Just as a heads up, I found that using the Ubuntu 20.04 image meant I also had to install cmake and clang-tidy, so it might be worth calling those out here.